data-model/src/main/scala/za/co/absa/enceladus/model/properties/propertyType/package.scala

```diff
@@ -28,16 +28,22 @@ package object propertyType {
     new Type(value = classOf[EnumPropertyType], name = "EnumPropertyType")
   ))
   sealed trait PropertyType {
-    def suggestedValue: String
+    def suggestedValue: Option[String]
 
     def isValueConforming(value: String): Try[Unit]
   }
 
-  case class StringPropertyType(suggestedValue: String = "") extends PropertyType {
+  case class StringPropertyType(suggestedValue: Option[String] = None) extends PropertyType {
     override def isValueConforming(value: String): Try[Unit] = Success(Unit)
   }
 
-  case class EnumPropertyType(allowedValues: Set[String], suggestedValue: String) extends PropertyType {
+  case class EnumPropertyType(allowedValues: Seq[String], suggestedValue: Option[String]) extends PropertyType {
+    require(allowedValues.nonEmpty, "At least one allowed value must be defined for EnumPropertyType.")
+    require(allowedValues.forall(_.nonEmpty), "Empty string is disallowed as an allowedValue for EnumPropertyType." +
+      "Consider using the property in non-mandatory context instead.")
+    require(allowedValues.size == allowedValues.toSet.size,
+      s"Allowed values for EnumPropertyType should be unique, but $allowedValues was given.")
```

Comment:
Question not stopping from releasing and would go into a new issue: 

How about printing only the duplicates?
Quick code option:
```scala
allowedValues
  .foldLeft(Map.empty[String, Int].withDefaultValue(0))((acc, v) => acc.updated(v, acc(v) + 1))
  .filter((m) => m._2 > 1)
  .keys
```

Reply:
👍 Let's do it with merge from Develop to Develop3